tracer/src/Datadog.Trace/Debugger/ExceptionAutoInstrumentation/ExceptionDebugging.cs

```diff
@@ -77,9 +77,16 @@ private static void InitSnapshotsSink()
                 uri => uri);
             var discoveryService = tracer.TracerManager.DiscoveryService;
             var gitMetadataTagsProvider = tracer.TracerManager.GitMetadataTagsProvider;
-            var batchApi = AgentBatchUploadApi.Create(apiFactory, discoveryService, gitMetadataTagsProvider);
-            var batchUploader = BatchUploader.Create(batchApi);
-            _sink = DebuggerSink.Create(snapshotStatusSink, new NopProbeStatusSink(), batchUploader, debuggerSettings);
+
+            var snapshotBatchUploadApi = AgentBatchUploadApi.Create(apiFactory, discoveryService, gitMetadataTagsProvider, false);
+            var snapshotBatchUploader = BatchUploader.Create(snapshotBatchUploadApi);
+
+            _sink = DebuggerSink.Create(
+                snapshotSink: snapshotStatusSink,
+                probeStatusSink: new NopProbeStatusSink(),
+                snapshotBatchUploader: snapshotBatchUploader,
+                diagnosticsBatchUploader: new NonBatchUploader(),
+                debuggerSettings);
```

Comment:
All these noop instancess feel like a smell to me. Feels like the DebuggerSink should be split in two or something? 🤔

Reply:
It's indeed not the cleanest. It will be addressed separately when we start clean up the Live Debugger code and initialize it correctly, as mentioned by you in https://github.com/DataDog/dd-trace-dotnet/pull/5163#discussion_r1505982833.

Reply:
Agree. We should do this also for `NopProbeStatusSink`, IMO.